### PR TITLE
#4 Jtj 50-rspec-incorrect-missing

### PR DESCRIPTION
**Pull request by mla-wtag**
**Description** 
This branch was created to address several issues found in my last merged branch JTJ-50-Login-cancancan. 
- I modified the names from user and admin to user_role and admin_role, but neglected to update my RSpec tests accordingly. 
- I added the RSpec test for the image upload feature that handle cases of incorrect content_type and excessive image file size.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class User < ApplicationRecord
   has_one_attached :profile_photo
   validates_presence_of :first_name, :last_name, :employee_id, :date_of_birth, :joining_day, :designation, :role, message: ->(_, _) { I18n.t('validations.presence') }
   validates :email, presence: true, format: { with: /\A[^@\s]+@[^@\s]+\z/, message: ->(_, _) { I18n.t('validations.valid') } }
-  validates :profile_photo, presence: true, content_type: ['image/jpeg', 'image/png', 'image/jpg'], size: { less_than: 15.megabytes, message: I18n.t('validations.photo') }
+  validates :profile_photo, presence: true, content_type: { in: ['image/jpeg', 'image/png', 'image/jpg'], message: I18n.t('validations.content_type') }, size: { less_than: 5.megabytes, message: I18n.t('validations.photo') }
```
Comment (evk-wtag) [inline app/models/user.rb]:
```suggestion
  validates :profile_photo, presence: true, content_type: %w(image/jpeg image/png image/jpg), message: I18n.t('validations.content_type') }, size: { less_than: 5.megabytes, message: I18n.t('validations.photo') }
```

File: spec/factories/users.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@
     designation { FFaker::Job.title }
     email { FFaker::Internet.email }
     profile_photo { Rack::Test::UploadedFile.new(Rails.root.join('spec', 'fixtures', 'test.jpeg'), 'image/jpeg') }
-    role { %w[user admin].sample }
+    role { %w[user_role admin_role].sample }
```
Comment (evk-wtag) [inline spec/factories/users.rb:11]: The role suffix was not really required 😅

File: spec/models/users_spec.rb [hunk 3]
```diff
@@ -44,4 +44,19 @@
...
+      expect(user.errors.full_messages).to include("#{I18n.t('attributes.profile_photo')} #{I18n.t('validations.content_type')}")
+      #expect(user.errors.full_messages).to include('Profile Photo has an invalid content type')
+    end
+  end
+
+  it 'validates size of profile_photo' do
+    user = build(:user, profile_photo: Rack::Test::UploadedFile.new(Rails.root.join('spec', 'fixtures', 'large_image.png'), 'image/png'))
+    expect(user).not_to be_valid
+    expect(user.errors.full_messages).to include("#{I18n.t('attributes.profile_photo')} #{I18n.t('validations.photo')}")
+  end
```
Comment (evk-wtag) [inline spec/models/users_spec.rb:61]: Let's add specs for the content types if not added previously

File: spec/models/users_spec.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@
   end
```
Comment (evk-wtag) [inline spec/models/users_spec.rb:20]: Let's add specs for the content types if not added previously

File: config/locales/en.yml [hunk 5]
```diff
@@ -34,7 +34,8 @@ en:
   validations: 
     presence: can't be blank
     valid:  must be valid
-    photo: is to large (max. is 15 Megabytes)
+    content_type: has an invalid content type
+    photo: is to large (max. is 5 Megabytes)
```
Comment (lra-wtag) [inline config/locales/en.yml]:
```suggestion
    photo: is too large (max. is 5 Megabytes)
```

File: spec/models/users_spec.rb [hunk 6]
```diff
@@ -44,4 +44,19 @@
     )
     expect(user2).not_to be_valid
   end
+
+  describe 'profile_photo validations' do
+    it 'validates content type of profile_photo' do
+      user = build(:user, profile_photo: Rack::Test::UploadedFile.new(Rails.root.join('spec', 'fixtures', 'test.txt'), 'text/plain'))
+      expect(user).not_to be_valid
+      expect(user.errors.full_messages).to include("#{I18n.t('attributes.profile_photo')} #{I18n.t('validations.content_type')}")
+      #expect(user.errors.full_messages).to include('Profile Photo has an invalid content type')
```
Comment (lra-wtag) [inline spec/models/users_spec.rb]: lets remove this line if not needed

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: spec/models/users_spec.rb (hunk 4 above)
Comment (evk-wtag) [reply spec/models/users_spec.rb:20]: it { is_expected.to validate_content_type_of(:profile_photo).allowing('image/jpg', 'image/jpeg', 'image/png') }

**Review by evk-wtag — APPROVED** [review]

**Review by lra-wtag — APPROVED** [review]

**Pull request merged**